services/motion/explore/explore.go

```diff
@@ -0,0 +1,575 @@
+	fsService   framesystem.Service
+	components  map[resource.Name]resource.Resource
+	services    map[resource.Name]resource.Resource
+	logger      golog.Logger
+	lock        sync.Mutex
+
+	obstacleResponseChan  chan moveResponse
+	executionResponseChan chan moveResponse
+	backgroundWorkers     *sync.WaitGroup
+}
+
+func (ms *explore) MoveOnMap(
+	ctx context.Context,
+	componentName resource.Name,
+	destination spatialmath.Pose,
+	slamName resource.Name,
+	extra map[string]interface{},
+) (bool, error) {
+	return false, errUnimplemented
+}
+
+func (ms *explore) MoveOnGlobe(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *geo.Point,
+	heading float64,
+	movementSensorName resource.Name,
+	obstacles []*spatialmath.GeoObstacle,
+	motionCfg *motion.MotionConfiguration,
+	extra map[string]interface{},
+) (bool, error) {
+	return false, errUnimplemented
+}
+
+func (ms *explore) MoveOnGlobeNew(
+	ctx context.Context,
+	req motion.MoveOnGlobeReq,
+) (string, error) {
+	return "", errUnimplemented
+}
+
+func (ms *explore) GetPose(
+	ctx context.Context,
+	componentName resource.Name,
+	destinationFrame string,
+	supplementalTransforms []*referenceframe.LinkInFrame,
+	extra map[string]interface{},
+) (*referenceframe.PoseInFrame, error) {
+	return nil, errUnimplemented
+}
+
+func (ms *explore) StopPlan(
+	ctx context.Context,
+	req motion.StopPlanReq,
+) error {
+	return errUnimplemented
+}
+
+func (ms *explore) ListPlanStatuses(
+	ctx context.Context,
+	req motion.ListPlanStatusesReq,
+) ([]motion.PlanStatusWithID, error) {
+	return nil, errUnimplemented
+}
+
+func (ms *explore) PlanHistory(
+	ctx context.Context,
+	req motion.PlanHistoryReq,
+) ([]motion.PlanWithStatus, error) {
+	return nil, errUnimplemented
+}
+
+func (ms *explore) Close(ctx context.Context) error {
+	utils.FlushChan(ms.obstacleResponseChan)
+	utils.FlushChan(ms.executionResponseChan)
+	ms.backgroundWorkers.Wait()
+	return nil
+}
+
+// Move takes a goal location and will plan and execute a movement to move a component specified by its name
+// to that destination.
+func (ms *explore) Move(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *referenceframe.PoseInFrame,
+	worldState *referenceframe.WorldState,
+	constraints *servicepb.Constraints,
+	extra map[string]interface{},
+) (bool, error) {
+	operation.CancelOtherWithLabel(ctx, exploreOpLabel)
+
+	// Parse extras
+	motionCfg, err := parseMotionConfig(extra)
+	if err != nil {
+		return false, err
+	}
+
+	// obstacleDetectors
+	obstacleDetectors, err := ms.createObstacleDetectors(motionCfg)
+	if err != nil {
+		return false, err
+	}
+
+	// Create kinematic base
+	kb, err := ms.createKinematicBase(ctx, componentName, motionCfg)
+	if err != nil {
+		return false, err
+	}
+
+	// Create motionplan plan
+	planInputs, err := ms.createMotionPlan(ctx, kb, destination.Pose(), worldState, extra)
+	if err != nil {
+		return false, err
+	}
+	var plan motionplan.Plan
+	for _, inputs := range planInputs {
```

Comment:
we don't need to do this if you return the value of `PlanMotion`. Here, in my understanding, we are undoing what `GetFrameSteps` does

Reply:
Removed!